tubular/imputers.py

```diff
@@ -461,40 +463,41 @@ def fit(self, X: pd.DataFrame, y: pd.Series) -> pd.DataFrame:
 
         super().fit(X, y)
 
-        n_nulls = y.isna().sum()
+        n_nulls = y.is_null().sum()
 
         if n_nulls > 0:
             msg = f"{self.classname()}: y has {n_nulls} null values"
             raise ValueError(msg)
 
         self.impute_values_ = {}
 
-        X_y = self._combine_X_y(X, y)
+        X_y = nw.from_native(self._combine_X_y(X, y))
         response_column = "_temporary_response"
 
         for c in self.columns:
-            c_nulls = X[c].isna()
+            c_nulls = X[c].is_null()
```

Comment:
S: following adam's lead to push for polars-y syntax, think this may be faster with:
`c_nulls=X.select(nw.col(c).is_null())`

Reply:
Ahh to be fair that doesn't actually work as it returns a df

Reply:
So this works, but for some reason adding .to_series() (Polars method) on the end doesn't work?

Reply:
interesting, what's the error like when you add .to_series?

Reply:
I've added this (kinda) - this returns a df when we need a series. I've tried chaining the .to_series() method but for some reason that throws an error? I've trialled the exact thing in a notebook without issue. Not sure if this is polars-y enough

Reply:
[here](https://github.com/lvgig/tubular/pull/342/commits/01de73607f1ef09031fb121b17b6b4bdc36c9212)

Reply:
Sorry my comments are all a bit delayed as I added them then got sidetracked before committing! It basically says that the dataframe doesn't have a 'to_series()' method

Reply:
ah OK weird - as there isn't a 'LazySeries' in polars the optimisations might be limited anyway, thanks for looking i'll approve!